assert/assertions_test.go

```diff
@@ -2794,14 +2794,19 @@ func TestNeverFalse(t *testing.T) {
 	True(t, Never(t, condition, 100*time.Millisecond, 20*time.Millisecond))
 }
 
+// TestNeverTrue checks Never with a condition that returns true on second call.
 func TestNeverTrue(t *testing.T) {
 	mockT := new(testing.T)
-	state := 0
+
+	// A synchronized counter using a channel
+	counter := make(chan int, 2)
+	counter <- 1
+	counter <- 2
+	defer close(counter)
```

Comment:
This channel is absolutely a solid fix. :+1: But I think a chan of bools is just more clear. And you don't need to defer the close().
```suggestion
	returns := make(chan bool, 2)
	returns <- false
	returns <- true
	close(returns)
```

Reply:
Oh, yes!
I'm fixing.

Reply:
Fixed and rebased.